react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,99 @@
-import React from 'react';
+
+import React, {Component} from 'react';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [{ squares: Array(9).fill(null)}],
+    stepNumber: 0,
+    xIsNext: true
+  }
+
+  handleClick = (i) => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length -1];
+    const squares = current.squares.slice();
+    if (this.calculateWinner(current.squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{squares: squares}]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
```

Comment:
When you update the state with a value in the state, try to use this:

this.setState(prevState => ({ 
  history: history.concat([{squares: squares}]),
  stepNumber: history.length,
   xIsNext: !prevState.xIsNext
})